### PR TITLE
Add Alvarion support

### PR DESCRIPTION
**Pull request by MajesticFalcon**
#546

### CODE REVIEW

**Comment by ytti** [comment]
you're requiring timeout, but you're not using it

for style can you move the require block for net/tftp after the other requires, not inside connect.


Then we're good to go. You could do git rebase to squas the commits 7 commits into single coherent commit, but that's no biggy, i'll merger regardless once the requires are done. Thanks!

File: lib/oxidized/input/tftp.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+module Oxidized
+  require 'timeout'
+  require 'stringio'
+  require_relative 'cli'
+
+  class TFTP < Input
+    RescueFail = {
```
Comment (danilopopeye) [inline lib/oxidized/input/tftp.rb]: Do we need this constant if nothing is selected?

File: lib/oxidized/input/tftp.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def output
+      ""
+    end
+
+    private
+
+    def disconnect
+      # TFTP uses UDP, there is no connection to close
+    #rescue Errno::ECONNRESET, IOError
```
Comment (danilopopeye) [inline lib/oxidized/input/tftp.rb]: Can we remove this?

File: lib/oxidized/model/alvarion [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Alvarion < Oxidized::Model
+
+  # Used in Alvarion wisp equipment
+
+  # Run this command as an instance of Model so we can access node
+  pre do
```
Comment (danilopopeye) [inline lib/oxidized/model/alvarion]: wow, this seems super hacky 😝

File: lib/oxidized/input/tftp.rb (hunk 2 above)
Comment (MajesticFalcon) [inline lib/oxidized/input/tftp.rb]: I was not comfortable removing commands that I wasn't sure their intent. I will remove them

Comment (danilopopeye) [inline lib/oxidized/input/tftp.rb]: You could also remove the 'ensure' here, better than a empty method

File: lib/oxidized/model/alvarion (hunk 3 above)
Comment (ytti) [inline lib/oxidized/model/alvarion]: maybe we should have more descriptive alias for pre or post, which would have same behaviour for now, just use-cases where argument themselves need to see instance context.

**Pull request merged**